File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/mecanumdrive/commands/RobotCentricCommand.java
package org.firstinspires.ftc.teamcode.subsystems.mecanumdrive.commands;

import androidx.annotation.NonNull;

import com.arcrobotics.ftclib.command.CommandBase;
import org.firstinspires.ftc.teamcode.subsystems.mecanumdrive.MecanumDriveSubsystem;
import java.util.function.DoubleSupplier;

public final class RobotCentricCommand extends CommandBase {
        @NonNull
        private final MecanumDriveSubsystem mecanumDriveSubsystem;

        @NonNull
        private final DoubleSupplier driveSupplier, strafeSupplier, turnSupplier;

        public RobotCentricCommand(
                @NonNull MecanumDriveSubsystem mecanumDriveSubsystem,
                @NonNull DoubleSupplier driveSupplier,
                @NonNull DoubleSupplier strafeSupplier,
                @NonNull DoubleSupplier turnSupplier
        ) {
            this.mecanumDriveSubsystem = mecanumDriveSubsystem;
            this.driveSupplier         = driveSupplier;
            this.strafeSupplier        = strafeSupplier;
            this.turnSupplier          = turnSupplier;

            addRequirements(mecanumDriveSubsystem);
        }

        @Override public void execute() {
            mecanumDriveSubsystem.robotCentric(
                    driveSupplier.getAsDouble(),
                    strafeSupplier.getAsDouble(),
                    turnSupplier.getAsDouble()
            );
        }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/mecanum/commands/RobotCentricCommand.kt
package org.firstinspires.ftc.teamcode.subsystems.mecanum.commands

import com.arcrobotics.ftclib.command.CommandBase
import org.firstinspires.ftc.teamcode.subsystems.mecanum.MecanumDrive
import java.util.function.DoubleSupplier
import kotlin.math.abs

class RobotCentricCommand(
    val mecanumDriveSubsystem: MecanumDrive,
    val driveSupplier: DoubleSupplier,
    val strafeSupplier: DoubleSupplier,
    val turnSupplier: DoubleSupplier,
) : CommandBase() {

    init { addRequirements(mecanumDriveSubsystem) }

    override fun execute() {
        var driveValue = driveSupplier.asDouble
        var strafeValue = strafeSupplier.asDouble
        var turnValue = turnSupplier.asDouble

        if (abs(driveValue) < 0.05) driveValue = 0.0
        if (abs(strafeValue) < 0.05) strafeValue = 0.0
        if (abs(turnValue) < 0.05) turnValue = 0.0

        mecanumDriveSubsystem.robotCentric(driveValue, strafeValue, turnValue)
    }
}
